restcomm/restcomm.testsuite/src/test/java/org/mobicents/servlet/restcomm/http/EndpointTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * TeleStax, Open Source Cloud Communications
+ * Copyright 2011-2016, Telestax Inc and individual contributors
+ * by the @authors tag.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation; either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+package org.mobicents.servlet.restcomm.http;
+
+import com.sun.jersey.api.client.Client;
+import com.sun.jersey.api.client.filter.HTTPBasicAuthFilter;
+import org.jboss.arquillian.container.test.api.Deployer;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.junit.After;
+
+import java.net.URL;
+
+/**
+ * Base class that hides some of the boilerplate code need to set-up a new automated test. You will need to extend
+ * and provide your own 'logger' and createWebArchiveNoGw() method.
+ *
+ * @author Orestis Tsakiridis
+ */
+public class EndpointTest {
+
+    protected static final String version = org.mobicents.servlet.restcomm.Version.getVersion();
+
+    @ArquillianResource
+    private Deployer deployer;
+    @ArquillianResource
+    URL deploymentUrl;
+
+    @After
+    public void after() throws InterruptedException {
+        Thread.sleep(1000);
+    }
+
+    protected String getResourceUrl(String suffix) {
+        String urlString = deploymentUrl.toString();
+        if ( urlString.endsWith("/") )
+            urlString = urlString.substring(0,urlString.length()-1);
+
+        if ( suffix != null && !suffix.isEmpty()) {
+            if (!suffix.startsWith("/"))
+                suffix = "/" + suffix;
+            return urlString + suffix;
+        } else
+            return urlString;
+
+    }
+
+    protected Client getClient(String username, String password) {
+        Client jersey = Client.create();
```

Comment:
have you considered using specific JUnit lib for REST testing(ie https://github.com/jayway/rest-assured)?